src/libs/SidebarUtils.js

```diff
@@ -62,6 +64,13 @@ Onyx.connect({
         }
         const reportID = CollectionUtils.extractCollectionItemID(key);
         lastReportActions[reportID] = _.last(_.toArray(actions));
+
+        // The report is only visible if it is the last action not deleted that
+        // does not match a closed or created state.
+        visibleReportActionItems[reportID] = _.first(_.filter(
+            ReportActionsUtils.getSortedReportActionsForDisplay(_.toArray(actions)),
+            action => (action.actionName !== CONST.REPORT.ACTIONS.TYPE.CREATED),
+        ));
```

Comment:
I agree! It's something that definitely can be optimized. Even `getSortedReportActionsForDisplay` can be optimized so that the 2 filters happening can be merged together and then sorted.

Reply:
I've made a commit with this fix, please let me know if this was what you meant